subprojects/core/src/main/java/org/gradle/process/internal/DefaultJavaForkOptions.java

```diff
@@ -210,9 +210,8 @@ public JavaForkOptions copyTo(JavaForkOptions target) {
         super.copyTo(target);
         options.copyTo(target);
         if (jvmArgumentProviders != null) {
-            for (CommandLineArgumentProvider jvmArgumentProvider : jvmArgumentProviders) {
-                target.jvmArgs(jvmArgumentProvider.asArguments());
-            }
+            target.getJvmArgumentProviders().clear();
```

Comment:
I think it is only using `jvmArgs` because `options.copyTo` will already clear the target jvm args: https://github.com/gradle/gradle/blob/917a6278dc77c6c64aadec1bd88f28259f9a20a9/subprojects/core/src/main/java/org/gradle/process/internal/JvmOptions.java#L342

Reply:
Ah, I see.  Of course, if something had been added to `target.jvmArgumentProviders` prior to calling this, I don't think it gets cleared out (unless I'm missing something) so previously the effective jvm arguments in the target would become the previous list of argument providers + the copied argument providers (added to jvmArgs).  With your change, it would now only become the copied argument providers.

I'm fine with saying "this was a defect and not the intended behavior which we're fixing now".  I just wanted to point out that this does look like a behavior change, potentially not covered by a test, and make sure we were making this change deliberately.